zebra-state/src/service/finalized_state.rs

```diff
@@ -838,7 +838,8 @@ impl FinalizedState {
 // up automatically eventually.
 impl Drop for FinalizedState {
     fn drop(&mut self) {
-        self.delete_ephemeral()
+        self.db.cancel_all_background_work(false);
+        self.delete_ephemeral();
     }
```

Comment:
Low priority - only if the rest of the PR works:
* we might want to add a log before we wait for the database to shut down, so the user knows what is happening

(We already log before we delete files in `delete_ephemeral`, so we don't need to do anything about the file delete wait here.)